datafusion/core/src/datasource/provider.rs

```diff
@@ -161,20 +161,73 @@ pub trait TableProvider: Sync + Send {
     /// Specify if DataFusion should provide filter expressions to the
     /// TableProvider to apply *during* the scan.
     ///
+    /// Some TableProviders can evaluate filters more efficiently than the
+    /// `Filter` operator in DataFusion, for example by using an index.
+    /// 
     /// The return value must have one element for each filter expression passed
     /// in. The value of each element indicates if the TableProvider can apply
-    /// that particular filter during the scan.
+    /// that particular filter during the scan. The position in the return value
+    /// Vec corresponds to the expression in the `filters` input. 
+    /// 
+    /// Here is an example of how this can be done:
+    /// 
+    /// ```rust
+    /// use datafusion::error::{Result, DataFusionError};
+    /// use datafusion_expr::{Expr, TableProviderFilterPushDown};
+    /// 
+    /// // Override the supports_filters_pushdown to evaluate which expressions 
+    /// // to accept as pushdown predicates.
+    /// fn supports_filters_pushdown(&self, filters: &[&Expr]) -> Result<Vec<TableProviderFilterPushDown>> {
```

Comment:
This doesn't compile for me when I run `cargo test --doc` locally and is also failing in CI. One issue here is that this function needs to be part of an `impl` block for a struct rather than a standalone function, otherwise there is no `self` to refer to. I also added some comments for smaller issues that caused compilation issues locally.

Reply:
Thanks @andygrove , all good catches.

To add the `impl` it would be for `TableProvider` so do I need to create an entire struct, etc. to do that? Any help on how to contribute a minimal doc example would be great. I use VS Code with Rust Analyzer, I also do not see an easy way to copy code into the doc and make it a fn doc comment.

Any help appreciated, then I'll get it cleaned up. I see other areas I can help with the docs too once I have a fully verified example.

Reply:
@alamb already said this, but you can prefix lines of code with `#` to still have them compile, but not appear in the generated docs.

See https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html#hiding-portions-of-the-example for more info